### PR TITLE
New handle endpoint and improvements

### PR DESCRIPTION
**Pull request by herrmannplatz**

### CODE REVIEW

File: test/unit_tests/user/UserRepositorySpec.coffee [hunk 1]
```diff
@@ -254,7 +254,11 @@ describe 'z.user.UserRepository', ->
 
       it 'resolves with username when username is not taken', (done) ->
         usernames = ['john_doe']
-        server.respondWith 'GET', "#{test_factory.settings.connection.rest_url}/users?handles=#{usernames[0]}", [404, {}, '']
+        server.respondWith 'GET', "#{test_factory.settings.connection.rest_url}/users?handles=#{usernames[0]}", [
+          200,
+          {'Content-Type': 'application/json'},
+          JSON.stringify(usernames)
```
Comment (lipis) [inline test/unit_tests/user/UserRepositorySpec.coffee]:
```coffescript
           200
           'Content-Type': 'application/json'
           JSON.stringify usernames
```

File: test/unit_tests/user/UserRepositorySpec.coffee [hunk 2]
```diff
@@ -267,7 +271,7 @@ describe 'z.user.UserRepository', ->
         server.respondWith 'GET', "#{test_factory.settings.connection.rest_url}/users?handles=#{usernames[0]}", [
           200,
           {'Content-Type': 'application/json'},
-          JSON.stringify([{handle: usernames[0]}])
+          JSON.stringify([])
```
Comment (lipis) [inline test/unit_tests/user/UserRepositorySpec.coffee]:
```coffeescript
          200
          'Content-Type': 'application/json'
          JSON.stringify []
```

File: test/unit_tests/user/UserRepositorySpec.coffee [hunk 3]
```diff
@@ -254,20 +254,24 @@ describe 'z.user.UserRepository', ->
 
       it 'resolves with username when username is not taken', (done) ->
         usernames = ['john_doe']
-        server.respondWith 'GET', "#{test_factory.settings.connection.rest_url}/users?handles=#{usernames[0]}", [404, {}, '']
+        server.respondWith 'POST', "#{test_factory.settings.connection.rest_url}/users/handles", [
+          200,
+          {'Content-Type': 'application/json'},
```
Comment (lipis) [inline test/unit_tests/user/UserRepositorySpec.coffee]: the brackets could be also go away.. but regardless of that the commas are not needed..

File: test/unit_tests/user/UserRepositorySpec.coffee [hunk 4]
```diff
@@ -254,20 +254,24 @@ describe 'z.user.UserRepository', ->
...
+          expect(_usernames).toEqual usernames
           done()
         .catch done.fail
 
       it 'rejects when username is taken', (done) ->
         usernames = ['john_doe']
-        server.respondWith 'GET', "#{test_factory.settings.connection.rest_url}/users?handles=#{usernames[0]}", [
+        server.respondWith 'POST', "#{test_factory.settings.connection.rest_url}/users/handles", [
           200,
           {'Content-Type': 'application/json'},
```
Comment (lipis) [inline test/unit_tests/user/UserRepositorySpec.coffee]: same here.. if the change was done above

File: test/unit_tests/user/UserRepositorySpec.coffee (hunk 3 above)
Comment (lipis) [reply test/unit_tests/user/UserRepositorySpec.coffee]: no hard feelings with the `{}`, but I think it's cleaner without them :)

File: test/unit_tests/user/UserRepositorySpec.coffee [hunk 5]
```diff
@@ -254,20 +254,24 @@ describe 'z.user.UserRepository', ->
 
       it 'resolves with username when username is not taken', (done) ->
         usernames = ['john_doe']
-        server.respondWith 'GET', "#{test_factory.settings.connection.rest_url}/users?handles=#{usernames[0]}", [404, {}, '']
+        server.respondWith 'POST', "#{test_factory.settings.connection.rest_url}/users/handles", [
+          200,
+          'Content-Type': 'application/json',
```
Comment (lipis) [inline test/unit_tests/user/UserRepositorySpec.coffee:259]: le commas: `,`

Comment (herrmannplatz) [reply test/unit_tests/user/UserRepositorySpec.coffee:259]: AHHHHHHHHHHHHHHH

**Pull request merged**